File: app/src/main/java/com/lumiyaviewer/lumiya/res/text/AutoValue_DrawableTextParams.java
package com.lumiyaviewer.lumiya.res.text;

final class AutoValue_DrawableTextParams extends DrawableTextParams {
    private final int backgroundColor;
    private final String text;

    AutoValue_DrawableTextParams(String str, int i) {
        if (str == null) {
            throw new NullPointerException("Null text");
        }
        this.text = str;
        this.backgroundColor = i;
    }

    public int backgroundColor() {
        return this.backgroundColor;
    }

    public boolean equals(Object obj) {
        boolean z = true;
        if (obj == this) {
            return true;
        }
        if (!(obj instanceof DrawableTextParams)) {
            return false;
        }
        DrawableTextParams drawableTextParams = (DrawableTextParams) obj;
        if (!this.text.equals(drawableTextParams.text())) {
            z = false;
        } else if (this.backgroundColor != drawableTextParams.backgroundColor()) {
            z = false;
        }
        return z;
    }

    public int hashCode() {
        return ((this.text.hashCode() ^ 1000003) * 1000003) ^ this.backgroundColor;
    }

    public String text() {
        return this.text;
    }

    public String toString() {
        return "DrawableTextParams{text=" + this.text + ", " + "backgroundColor=" + this.backgroundColor + "}";
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/res/text/AutoValue_DrawableTextParams.kt
package com.lumiyaviewer.lumiya.res.text

class AutoValue_DrawableTextParams : DrawableTextParams {
    private Int backgroundColor
    private String text

    AutoValue_DrawableTextParams(String str, Int i) {
        if (str == null) {
            throw NullPointerException("Null text")
        }
        this.text = str
        this.backgroundColor = i
    }

    fun backgroundColor(): Int {
        return this.backgroundColor
    }

    fun equals(obj: Any): Boolean {
        if (obj == this) {
            return true
        }
        if (!(obj instanceof DrawableTextParams)) {
            return false
        }
        DrawableTextParams drawableTextParams = (DrawableTextParams) obj
        if (this.text == drawableTextParams.text()) {
            return this.backgroundColor == drawableTextParams.backgroundColor()
        }
        return false
    }

    fun hashCode(): Int {
        return ((this.text.hashCode() ^ 1000003) * 1000003) ^ this.backgroundColor
    }

    fun text(): String {
        return this.text
    }

    fun toString(): String {
        return "DrawableTextParams{text=" + this.text + ", " + "backgroundColor=" + this.backgroundColor + "}"
    }
}
